### PR TITLE
Updated the wavefront reporter, so that it can read the VCAP_SERVICES…

### PR DESCRIPTION
**Pull request by spramanik4667**
… env variable in PCF to determine proxy hostname and port.

### CODE REVIEW

File: dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java [hunk 1]
```diff
@@ -185,6 +187,69 @@ public Builder withJvmMetrics() {
     }
 
     /**
+     * Builds a {@link WavefrontReporter} from the VCAP_SERVICES env variable, sending metrics
+     * using the given {@link WavefrontSender}. This should be used in PCF environment only.
+     *
+     * @param proxyServiceName The name of the wavefront proxy service. If wavefront-tile is used
```
Comment (panghy) [inline dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java]: param comments should align (seems like you have a different code style)

File: dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java [hunk 2]
```diff
@@ -185,6 +187,69 @@ public Builder withJvmMetrics() {
...
+     *                        to deploy the proxy, then the service name will be 'wavefront-proxy'.
+     * @param failOnError A flag to determine what to do if the service parameters are not
+     *                    available. If 'true' then the method will throw RuntimeException else
+     *                    it will log an error message and continue.
+     * @return a {@link WavefrontReporter}
+     */
+    public WavefrontReporter bindToCloudFoundryService(String proxyServiceName,
```
Comment (panghy) [inline dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java]: add an overloaded method that assumes proxyServiceName is "wavefront-proxy"? (perhaps also overloaded methods where failOfError is false too?)

File: dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java [hunk 3]
```diff
@@ -185,6 +187,69 @@ public Builder withJvmMetrics() {
...
+                                                       boolean failOnError) {
+      String proxyHostname;
```
Comment (panghy) [inline dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java]: check that proxyServiceName is not null?

File: dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java (hunk 2 above)
Comment (panghy) [inline dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java]: also mark proxyServiceName as @Nonnull

File: dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java (hunk 1 above)
Comment (spramanik4667) [inline dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java]: Reformatted.

File: dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java (hunk 2 above)
Comment (spramanik4667) [inline dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java]: Added overloaded methods.

File: dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java (hunk 3 above)
Comment (spramanik4667) [inline dropwizard-metrics/3.1/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java]: Added a Preconditions NotNull check.

**Pull request merged**